File: backend/model/friend.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Friend struct {
	UserOneID  				int `gorm:"primaryKey"`
	UserTwoID 				int `gorm:"primaryKey"`
	CreatedAt 				time.Time
	DeletedAt 				gorm.DeletedAt
	Pending_First_Second		bool
	Pending_Second_First		bool
	Friends									bool
}

File: frontend/src/Types/Friend.ts
import { User } from "./User"

export interface IFriend {
  UserOneID: number,  				
  UserTwoID: number								
  Pending_First_Second: boolean		
  Pending_Second_First: boolean		
  Friends: boolean							
}

export interface IFriendWithUser {
  friendship: IFriend,
  friendUser: Partial<User>
}